keras_cv/layers/regularization/dropblock_2d.py

```diff
@@ -0,0 +1,191 @@
+        output = layer(features, training=True)
+
+        # Preview the feature map after dropblock:
+        print(output[..., 0])
+        # tf.Tensor(
+        # [[[0.10955477 0.54570675 0.5242462  0.42167106]
+        #   [0.46290365 0.97599393 0.75158674 1.3025614 ]
+        #   [0.         0.         0.39221907 0.6840355 ]
+        #   [0.         0.         0.80540895 0.14002927]]], shape=(1, 4, 4),
+        # dtype=float32)
+
+    ```
+        We can observe two things:
+        1. A 2x2 block has been set to zero.
+        2. The inputs have been scaled as mentioned in the paper.
+
+        One must remember, that DropBlock operation is random, so bigger or smaller
+        patches can be dropped.
+    """
+
+    def __init__(
+        self,
+        keep_probability=0.9,
+        dropblock_size=7,
+        data_format=None,
+        seed=None,
+        name=None,
+    ):
+        super().__init__(seed=seed, name=name, force_generator=True)
+        if not dropblock_size > 0:
+            raise ValueError(
+                f"dropblock_size must be greater than 0. Received: {dropblock_size}"
+            )
+        if not 0.0 <= keep_probability <= 1.0:
+            raise ValueError(
+                f"keep_probability must be a number between 0 and 1. "
+                f"Received: {keep_probability}"
+            )
+
+        self._keep_probability = keep_probability
+        self._dropblock_size = dropblock_size
+        self._data_format = conv_utils.normalize_data_format(data_format)
+
+    def call(self, x, training=False):
+        if not training or self._keep_probability == 1.0:
+            return x
+
+        if self._data_format == "channels_last":
+            _, height, width, _ = x.get_shape().as_list()
+        else:
+            _, _, height, width = x.get_shape().as_list()
+
+        total_size = width * height
+        dropblock_size = min(self._dropblock_size, width, height)
+
+        # Seed_drop_rate is the gamma parameter of DropBlock.
+        seed_drop_rate = (
+            (1.0 - self._keep_probability)
+            * total_size
+            / dropblock_size**2
+            / ((width - self._dropblock_size + 1) * (height - self._dropblock_size + 1))
+        )
+
+        # Forces the block to be inside the feature map.
+        w_i, h_i = tf.meshgrid(tf.range(width), tf.range(height))
+        valid_block = tf.logical_and(
+            tf.logical_and(
+                w_i >= int(dropblock_size // 2), w_i < width - (dropblock_size - 1) // 2
+            ),
+            tf.logical_and(
+                h_i >= int(dropblock_size // 2), h_i < width - (dropblock_size - 1) // 2
+            ),
+        )
+
+        if self._data_format == "channels_last":
+            valid_block = tf.reshape(valid_block, [1, height, width, 1])
+        else:
+            valid_block = tf.reshape(valid_block, [1, 1, height, width])
+
+        randnoise = self._random_generator.random_uniform(tf.shape(x), dtype=tf.float32)
```

Comment:
change to random_noise per the keras naming convention of not using abbreviations

Reply:
Changed to `random_noise`